cmake/SDL.cmake

```diff
@@ -71,7 +71,8 @@ if(UNIX)
     elseif("${CMAKE_CXX_COMPILER_ID}" STREQUAL "Clang")
         get_filename_component(CXX_CMD_NAME ${CMAKE_CXX_COMPILER} NAME)
         # Fujitsu CXX compiler does not support "-fstack-protector-all".
-        if(NOT CXX_CMD_NAME STREQUAL "FCC")
+        # The same applies to GPU targets (CUDA/HIP).
+        if(NOT CXX_CMD_NAME STREQUAL "FCC" AND NOT(DNNL_SYCL_CUDA OR DNNL_SYCL_HIP))
```

Comment:
It seems to me that CPU backend stack protection will be disabled as well with this change. Not sure this is the intention.

Reply:
How do you mean, previously it would skip adding the flag if not using `FCC`, now it check if we are not using `FCC` and are not compiling for either CUDA or HIP.
Perhaps a cleaner way would be to rewrite it to:
```cmake
if (NOT (CXX_CMD_NAME STREQUAL "FCC" OR DNNL_SYCL_CUDA OR DNN_SYCL_HIP))
```

Reply:
Why is this set in the default build config at all? Isn't it to the downstream user to decide whether to slow down their build for the extra security e.g. what Linux distributions do with setting these in their package build scripts et al

I personally wouldn't set this by default and leave it for folk to configure with `CXXFLAGS=` or similar mechanisms. You might see a few cycles shaved off too ;)

Reply:
I agree, but it seems like a discussion for another ticket. I'd like to fix the build for people targeting AMD as soon as possible, as it manifest through a hard compiler crash that might not be trivial for everyone to trace down.

I've simplified the boolean logic in the if clause.

Reply:
> Why is this set in the default build config at all? Isn't it to the downstream user to decide whether to slow down their build for the extra security e.g. what Linux distributions do with setting these in their package build scripts et al

It's enabled by default to meet security requirements. Having the right flags by default helps to fortify downstream applications.